server/aws-lsp-partiql/src/server/syntax-highlighting/parser-tokens.ts

```diff
@@ -0,0 +1,195 @@
+import Parser from 'web-tree-sitter'
+import parserBase64 from '../tree-sitter-parser/tree-sitter-parser-inline'
+import wasmBinaryArray from '../tree-sitter-parser/tree-sitter-inline'
+import {
+    SemanticTokenTypes,
+    SemanticTokens,
+    SemanticTokenModifiers,
+    Range,
+} from '@aws/language-server-runtimes/server-interface'
+import { SemanticTokensBuilder } from 'vscode-languageserver/node'
+import { semanticTokensLegend } from '../language-service'
+
+export interface SemanticToken {
+    range: Range
+    tokenType: SemanticTokenTypes
+    tokenLength: number
+    /**
+     * An optional array of modifiers for this token
+     */
+    tokenModifiers?: SemanticTokenModifiers[]
+}
+
+export const string2TokenTypes: { [key: string]: SemanticTokenTypes } = {
+    tuple_punc_start: SemanticTokenTypes.operator,
+    tuple_punc_end: SemanticTokenTypes.operator,
+    tuple_punc_separator: SemanticTokenTypes.operator,
+    pair_punc_separator: SemanticTokenTypes.operator,
+    array_punc_start: SemanticTokenTypes.operator,
+    array_punc_end: SemanticTokenTypes.operator,
+    array_punc_separator: SemanticTokenTypes.operator,
+    bag_punc_start: SemanticTokenTypes.operator,
+    bag_punc_end: SemanticTokenTypes.operator,
+    bag_punc_separator: SemanticTokenTypes.operator,
+    ion: SemanticTokenTypes.string,
+}
+
+// Initialize and prepare the parser
+async function initParser() {
+    let wasmModuleUrl: string | null = null
+    if (typeof window !== 'undefined' && typeof window.URL.createObjectURL === 'function') {
```

Comment:
We're missing the case here where the server is being run inside a [Web worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers). For web workers it could look something like

```typescript
  else if (typeof WorkerGlobalScope !== 'undefined' && self instanceof WorkerGlobalScope) {
      // Webworker
      const wasmBlob = new Blob([tree_sitter_inline_1.default], { type: 'application/wasm' });
      wasmModuleUrl = URL.createObjectURL(wasmBlob);
  }
```

Reply:
Thanks for pointing out. Added the Webworker Case in a latest commit. To avoid changing the code in `tsconfig.json`, use a type assertion instead here:
``` typescript
else if (typeof self !== 'undefined' && self.constructor.name === "DedicatedWorkerGlobalScope") {
    // Webworker
    const wasmBlob = new Blob([wasmBinaryArray], { type: 'application/wasm' })
    wasmModuleUrl = URL.createObjectURL(wasmBlob)
} 
```